### PR TITLE
feat:[RP-20] - typography with variants

### PR DESCRIPTION
**Pull request by dipSyxx**
# Title
feat:[RP-20] - typography with variants

## Ticket Link
https://rozetka-home-project.atlassian.net/browse/

### CODE REVIEW

File: src/elements/Typography/TypographyCustom.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable react/forbid-component-props */
```
Comment (MaksPapirovnyk) [inline src/elements/Typography/TypographyCustom.tsx:1]: Why there is this comment?

File: src/elements/Typography/TypographyCustom.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import React, { ReactNode } from 'react'
+
+type TypographyCustomProps = {
+  children: ReactNode
+  component: keyof JSX.IntrinsicElements
+  variant: string
```
Comment (MaksPapirovnyk) [inline src/elements/Typography/TypographyCustom.tsx:7]: `variant: string`
how to understand which variants exist? just pass any string?
probably needs variants mapping
https://blog.scottlogic.com/2021/07/27/union-types-react-components.html

File: src/elements/Typography/TypographyCustom.tsx (hunk 1 above)
Comment (dipSyxx) [reply src/elements/Typography/TypographyCustom.tsx:1]: because eslint gives error

File: src/elements/Typography/TypographyCustom.tsx (hunk 2 above)
Comment (dipSyxx) [reply src/elements/Typography/TypographyCustom.tsx]: fixed

File: src/elements/Typography/TypographyCustom.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable react/forbid-component-props */
+import React, { ReactNode } from 'react'
+
+type TypographyCustomProps = {
```
[the remark this answers is not in the document]
Comment (dipSyxx) [reply src/elements/Typography/TypographyCustom.tsx:5]: fixed

File: src/elements/Typography/TypographyCustom.tsx (hunk 1 above)
Comment (MaksPapirovnyk) [reply src/elements/Typography/TypographyCustom.tsx]: ahaha 😆

File: src/elements/Typography/typographyStyles.module.sass [hunk 4]
```diff
@@ -0,0 +1,13 @@
+.typography
+  font-weight: 400
+  line-height: 1.4
```
Comment (MaksPapirovnyk) [inline src/elements/Typography/typographyStyles.module.sass:3]: Please use em size

**Review by MaksPapirovnyk — APPROVED** [review]

**Pull request merged**